src/security/tests/acl_api_tests.c

```diff
@@ -0,0 +1,1433 @@
+
+	assert_int_equal(daos_acl_remove_ace_realloc(orig_acl,
+			DAOS_ACL_EVERYONE + 1, ace[0]->dae_principal,
+			ace[0]->dae_principal_len, &result_acl), -DER_INVAL);
+
+	assert_null(result_acl);
+
+	/* cleanup */
+	daos_acl_free(orig_acl);
+	free_all_aces(ace, num_aces);
+}
+
+static void
+test_acl_remove_ace_missing_name(void **state)
+{
+	int		num_aces = 1;
+	struct daos_ace	*ace[num_aces];
+	struct daos_acl	*orig_acl;
+	struct daos_acl	*result_acl = NULL;
+
+	fill_ace_list_with_users(ace, num_aces);
+	orig_acl = daos_acl_alloc(ace, num_aces);
+
+	assert_int_equal(daos_acl_remove_ace_realloc(orig_acl,
+			DAOS_ACL_USER, NULL, 5, &result_acl), -DER_INVAL);
+	assert_null(result_acl);
+
+	assert_int_equal(daos_acl_remove_ace_realloc(orig_acl,
+			DAOS_ACL_GROUP, NULL, 5, &result_acl), -DER_INVAL);
+	assert_null(result_acl);
+
+	/* cleanup */
+	daos_acl_free(orig_acl);
+	free_all_aces(ace, num_aces);
+}
+
+static void
+test_acl_remove_ace_name_len_zero(void **state)
+{
+	int		num_aces = 1;
+	struct daos_ace	*ace[num_aces];
+	struct daos_acl	*orig_acl;
+	struct daos_acl	*result_acl = NULL;
+
+	fill_ace_list_with_users(ace, num_aces);
+	orig_acl = daos_acl_alloc(ace, num_aces);
+
+	assert_int_equal(daos_acl_remove_ace_realloc(orig_acl,
+			DAOS_ACL_USER, "user1@", 0, &result_acl), -DER_INVAL);
+	assert_null(result_acl);
+
+	assert_int_equal(daos_acl_remove_ace_realloc(orig_acl,
+			DAOS_ACL_GROUP, "group1@", 0, &result_acl), -DER_INVAL);
+	assert_null(result_acl);
+
+	/* cleanup */
+	daos_acl_free(orig_acl);
+	free_all_aces(ace, num_aces);
+}
+
+static void
+test_acl_remove_ace_one_user(void **state)
+{
+	int		num_aces = 1;
+	struct daos_ace	*ace[num_aces];
+	struct daos_acl	*orig_acl;
+	struct daos_acl	*result_acl = NULL;
+
+	fill_ace_list_with_users(ace, num_aces);
+	orig_acl = daos_acl_alloc(ace, num_aces);
+
+	assert_int_equal(daos_acl_remove_ace_realloc(orig_acl,
+			ace[0]->dae_principal_type, ace[0]->dae_principal,
+			ace[0]->dae_principal_len, &result_acl), DER_SUCCESS);
+
+	/* Result should be empty ACL */
+	assert_non_null(result_acl);
+	assert_int_equal(result_acl->dal_len, 0);
+	assert_ptr_not_equal(result_acl, orig_acl);
+
+	/* cleanup */
+	daos_acl_free(orig_acl);
+	daos_acl_free(result_acl);
+	free_all_aces(ace, num_aces);
+}
+
+static void
+test_acl_remove_ace_multi_user(void **state)
+{
+	int		num_aces = 4;
+	struct daos_ace	*ace[num_aces];
+	struct daos_acl	*orig_acl;
+	struct daos_acl	*result_acl = NULL;
+	int		removed_idx = 2;
+	int		i;
+
+	fill_ace_list_with_users(ace, num_aces);
+	orig_acl = daos_acl_alloc(ace, num_aces);
+
+	assert_int_equal(daos_acl_remove_ace_realloc(orig_acl,
+			ace[removed_idx]->dae_principal_type,
+			ace[removed_idx]->dae_principal,
+			ace[removed_idx]->dae_principal_len, &result_acl),
+			DER_SUCCESS);
+
+	/* Result should have only removed that user */
+	assert_non_null(result_acl);
+	assert_ptr_not_equal(result_acl, orig_acl);
+	assert_int_equal(result_acl->dal_len,orig_acl->dal_len -
```

Comment:
(style)  space required after that ',' (ctx:VxV)

Reply:
Fixed